### PR TITLE
on_action can be a function (#939)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
Addresses #939.

### CODE REVIEW

**Review by FabienLelaquais — APPROVED** [review]
Eaaaasy! 🙏

File: src/taipy/gui/gui.py [hunk 1]
```diff
@@ -1600,7 +1601,11 @@ def _is_broadcasting(self) -> bool:
         except RuntimeError:
             return False
 
-    def _download(self, content: t.Any, name: t.Optional[str] = "", on_action: t.Optional[str] = ""):
+    def _download(self, content: t.Any, name: t.Optional[str] = "", on_action: t.Optional[t.Union[str, t.Callable]] = ""):
+        if callable(on_action):
+            on_action_name = _get_expr_var_name(on_action.__name__)
```
Comment (dinhlongviolin1) [inline src/taipy/gui/gui.py]: on_action.__name__ might be undefined

**Review by dinhlongviolin1 — APPROVED** [review]
First review on new phone!

File: src/taipy/gui/gui.py (hunk 1 above)
Comment (FabienLelaquais) [reply src/taipy/gui/gui.py]: Didn't think of this one. Lambdas?

**Comment by FabienLelaquais** [comment]
> First review on new phone!

😜

File: src/taipy/gui/gui.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/gui.py]: When would that be ?

**Pull request merged**